src/v/cloud_topics/app.cc

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "cloud_topics/app.h"
+
+#include "cloud_topics/reconciler/reconciler.h"
+
+namespace experimental::cloud_topics::reconciler {
+
+app::app(
+  seastar::sharded<cluster::partition_manager>* partition_manager,
+  seastar::sharded<cloud_io::remote>* remote)
+  : _reconciler(std::make_unique<reconciler>(partition_manager, remote)) {}
+
+app::~app() = default;
+
+seastar::future<> app::start() {
+    vassert(!_reconciler, "cannot restart service");
```

Comment:
won't this always fire? did you mean to `make_unique` in here or something?

Reply:
oh good catch. yeh, make_unique is in the constructor. i just removed the assert.